src/pages/settings/components/checkbox.js

```diff
@@ -11,26 +11,36 @@
 
 import { html } from 'hybrids';
 
-function clickSlottedElement(host, event) {
-  const target = host.render().querySelector('slot').assignedElements()[0];
-  if (event.target !== target) {
-    target.click();
+function clickInput(host, event) {
+  if (host.input && event.target !== host.input) {
+    event.stopPropagation();
+    host.input.click();
   }
 }
 
 export default {
   disabled: { value: false, reflect: true },
+  input: (host) => host.querySelector('input'),
+  checked: {
+    value: (host) => host.input?.checked ?? false,
+    connect: (host, key, invalidate) => {
+      host.input?.addEventListener('change', invalidate);
+      return () => {
+        host.input?.removeEventListener('change', invalidate);
+      };
+    },
+  },
```

Comment:
why this is not a `onchange` property on the input? because the slot? It took me a while to understand what it does.

Reply:
The component is a UI wrapper for the `<input>` element, and it is in the `<slot>` element.

Reply:
ok, I like this version more than `const target = host.render().querySelector('slot').assignedElements()[0];`